com.unity.render-pipelines.core/Editor/Lighting/ProbeVolume/ProbeGIBaking.cs

```diff
@@ -626,6 +657,193 @@ static void OnLightingDataCleared()
+                    offsetsCount = bakingCell.offsetVectors.Length
+                };
+                asset.cellCounts[i] = cellCounts;
+                asset.totalCellCounts.Add(cellCounts);
+            }
+
+            // CellData
+            using var bricks = new NativeArray<Brick>(asset.totalCellCounts.bricksCount, Allocator.Persistent, NativeArrayOptions.UninitializedMemory);
+            using var probesL0L1 = new NativeArray<float>(asset.totalCellCounts.probesCount * ProbeVolumeAsset.kL0L1ScalarCoefficientsCount, Allocator.Persistent, NativeArrayOptions.UninitializedMemory);
+
+            // CellOptionalData
+            var probesL2ScalarPaddedCount = asset.bands == ProbeVolumeSHBands.SphericalHarmonicsL2 ? asset.totalCellCounts.probesCount * ProbeVolumeAsset.kL2ScalarCoefficientsCount + 3 : 0;
+            using var probesL2 = new NativeArray<float>(probesL2ScalarPaddedCount, Allocator.Persistent, NativeArrayOptions.UninitializedMemory);
+
+            // CellSupportData
+            using var positions = new NativeArray<Vector3>(asset.totalCellCounts.probesCount, Allocator.Persistent, NativeArrayOptions.UninitializedMemory);
+            using var validity = new NativeArray<float>(asset.totalCellCounts.probesCount, Allocator.Persistent, NativeArrayOptions.UninitializedMemory);
+            using var offsets = new NativeArray<Vector3>(asset.totalCellCounts.offsetsCount, Allocator.Persistent, NativeArrayOptions.UninitializedMemory);
+
+            var startCounts = new ProbeVolumeAsset.CellCounts();
+            for (var i = 0; i < bakingCells.Count; ++i)
+            {
+                var bakingCell = bakingCells[i];
+                var cellCounts = asset.cellCounts[i];
+
+                bricks.GetSubArray(startCounts.bricksCount, cellCounts.bricksCount).CopyFrom(bakingCell.bricks);
+
+                var probesTargetL0L1 = probesL0L1.GetSubArray(startCounts.probesCount * ProbeVolumeAsset.kL0L1ScalarCoefficientsCount, cellCounts.probesCount * ProbeVolumeAsset.kL0L1ScalarCoefficientsCount);
+                for (int j = 0, k = 0; j < cellCounts.probesCount; ++j, k += ProbeVolumeAsset.kL0L1ScalarCoefficientsCount)
+                {
+                    ref var sh = ref bakingCell.sh[j];
+                    WriteToShaderCoeffsL0L1(ref sh, probesTargetL0L1, k);
+                }
+
+                if (asset.bands == ProbeVolumeSHBands.SphericalHarmonicsL2)
+                {
+                    var probesTargetL2 = probesL2.GetSubArray(startCounts.probesCount * ProbeVolumeAsset.kL2ScalarCoefficientsCount, cellCounts.probesCount * ProbeVolumeAsset.kL2ScalarCoefficientsCount);
+                    for (int j = 0, k = 0; j < cellCounts.probesCount; ++j, k += ProbeVolumeAsset.kL2ScalarCoefficientsCount)
+                    {
+                        ref var sh = ref bakingCell.sh[j];
+                        WriteToShaderCoeffsL2(ref sh, probesTargetL2, k);
+                    }
+                }
```

Comment:
Shouldn't the L2 coefs be written regardless of the settings ?

Reply:
I would say no since I don't think we should force ppl to have L2 data. That being said, `asset.bands` is always `SphericalHarmonicsL2` now until we add an explicit option to only bake L0L1 data.

Reply:
ah thanks i missed that part